packages/@o3r/rules-engine/src/services/rules-engine.effect.ts

```diff
@@ -100,9 +118,13 @@ export class PlaceholderTemplateResponseEffect {
         }
       }
     }
-    return {renderedTemplate: template, unknownTypeFound};
+    return replacements$.length > 0 && !!template ?
+      combineLatest(replacements$).pipe(
+        map((replacements: { ejsVar: RegExp; value: string }[]) =>
+          replacements.reduce((acc, replacement) => ({
+            renderedTemplate: acc.renderedTemplate.replace(replacement.ejsVar, replacement.value),
+            unknownTypeFound: acc.unknownTypeFound
```

Comment:
this property is never touched during the iteration, the `reduce` could return the replaced string